airflow/jobs/scheduler_job.py

```diff
@@ -1307,20 +1309,27 @@ def _schedule_dag_run(
 
         return callback_to_run
 
-    def _verify_integrity_if_dag_changed(self, dag_run: DagRun, session: Session) -> None:
-        """Only run DagRun.verify integrity if Serialized DAG has changed since it is slow"""
+    def _verify_integrity_if_dag_changed(self, dag_run: DagRun, session: Session) -> bool:
+        """
+        Only run DagRun.verify integrity if Serialized DAG has changed since it is slow.
+
+        Return True if we determine that DAG still exists.
+        """
         latest_version = SerializedDagModel.get_latest_version_hash(dag_run.dag_id, session=session)
         if dag_run.dag_hash == latest_version:
             self.log.debug("DAG %s not changed structure, skipping dagrun.verify_integrity", dag_run.dag_id)
-            return
+            return True
 
         dag_run.dag_hash = latest_version
 
         # Refresh the DAG
         dag_run.dag = self.dagbag.get_dag(dag_id=dag_run.dag_id, session=session)
+        if not dag_run.dag:
+            return False
```

Comment:
Maybe this should be an exception (caught outside) instead; returning True and False doesn’t feel natural.

Or maybe we should refactor the code around this to be

```python
# This returns
# a. DAG is it changed.
# b. None if not changed.
# c. None if the DAG is not found in DagBag, with a log message.
refreshed_dag = self._get_changed_dag_()
if refreshed_dag is not None:
    dag_run.dag = refreshed_dag
    dag_run.verify_integrity(session=session)
```

Reply:
I think it's pretty goot to be like it is. Exceptions are fine, but sometimes True/False is fine - especially if the method is named as "verify..".